### PR TITLE
update external-snapshotter version to v4.1.0

### PR DESCRIPTION
**Pull request by agarwal-mudit**
Creating a wip request because I am trying to test these changes on an OCS cluster, but it can be reviewed in parallel.

### CODE REVIEW

**Comment by agarwal-mudit** [comment]
cc @Madhu-1

File: deploy/bundle/manifests/backingstore.crd.yaml [hunk 1]
```diff
@@ -30,10 +30,14 @@ spec:
         description: BackingStore is the Schema for the backingstores API
         properties:
           apiVersion:
-            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            description: 'APIVersion defines the versioned schema of this representation
```
Comment (Madhu-1) [inline deploy/bundle/manifests/backingstore.crd.yaml:33]: snapshotter changes look good to me but these are unrelated to this PR. do we need a separate PR for this one?

File: go.mod [hunk 2]
```diff
@@ -6,9 +6,13 @@ require (
...
-	github.com/go-logr/logr v0.3.0
-	github.com/imdario/mergo v0.3.10
-	github.com/kubernetes-csi/external-snapshotter/v2 v2.1.2
+	github.com/go-logr/logr v0.4.0
+	github.com/gogo/protobuf v1.3.2 // indirect
+	github.com/google/go-cmp v0.5.4 // indirect
+	github.com/google/gofuzz v1.2.0 // indirect
+	github.com/googleapis/gnostic v0.5.3 // indirect
+	github.com/imdario/mergo v0.3.11
+	github.com/kubernetes-csi/external-snapshotter/client/v4 v4.0.0
```
Comment (Madhu-1) [inline go.mod:15]: this should be 4.1.0? which is used with    OCP 4.8

File: deploy/bundle/manifests/backingstore.crd.yaml (hunk 1 above)
Comment (agarwal-mudit) [reply deploy/bundle/manifests/backingstore.crd.yaml:33]: This is automatic after running `make gen-latest-csv`

File: go.mod (hunk 2 above)
Comment (agarwal-mudit) [reply go.mod:15]: Good catch, looks like something happened after running `go mod vendor`, let me check

File: deploy/bundle/manifests/backingstore.crd.yaml (hunk 1 above)
Comment (agarwal-mudit) [reply deploy/bundle/manifests/backingstore.crd.yaml]: This is also not required after latest commits.

**Comment by Madhu-1** [comment]
LGTM

**Comment by agarwal-mudit** [comment]
Build changes for sidecar updates are merged and we know that this will not cause UI regression hence removed the WIP label.

@jarrpa @umangachapagain PTAL

**Review by jarrpa — APPROVED** [review]
/lgtm
